voila/handler.py

```diff
@@ -16,6 +17,16 @@
 from .exporter import VoilaExporter
 
 
+def update_nested_dict(d, u):
+    """Update (nested) dictionaries: update keys down to any depth."""
+    for k, v in u.items():
+        if isinstance(v, collections.Mapping):
+            d[k] = update_nested_dict(d.get(k, {}), v)
+        else:
+            d[k] = v
+    return d
```

Comment:
Can we use https://github.com/jupyter/jupyter_server/blob/master/jupyter_server/config_manager.py#L19 ?

Reply:
Indeed I did not think of the config manager merging mechanism.

@maartenbreddels can you push to this branch making that change?

Reply:
@maartenbreddels Thank you! A helper function to update dictionaries recursively had to exist 'nearby' indeed...

@SylvainCorlay Done via 6ea5bc7 (function call changed because the config manager's function is in-place).

Reply:
Great! I happen to know because I worked on that module :)